oonipipeline/src/oonipipeline/transforms/measurement_transformer.py

```diff
@@ -359,6 +338,44 @@ def find_tls_handshake_network_events(
     return None
 
 
+def find_tls_handshake_network_events_with_transaction_id(
+    tls_handshake: TLSHandshake,
+    network_events: Optional[List[NetworkEvent]],
+) -> Optional[List[NetworkEvent]]:
+    transaction_map = {}
+    all_event_windows_dict = {}
+
+    for ne in network_events:
+        if ne.transaction_id is None:
+            log.warning("detected network event without transaction_id")
+            continue
+        
+        ne_addr = ne.address
+        if ne_addr not in all_event_windows_dict:
+            all_event_windows_dict[ne_addr] = defaultdict(list)
+            if not transaction_map[ne.transaction_id]:
+                transaction_map[ne.transaction_id] = True
+            else:
+                log.warning("detected same transactions with different addresses")
```

Comment:
Should we maybe add some additional context here like the index at which it failed and some metadata about the failing transaction so we can debug it easier?